warehouse/query-core/src/test/java/datawave/query/function/IndexOnlyKeyToDocumentDataTest.java

```diff
@@ -0,0 +1,148 @@
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iteratorsImpl.system.SortedMapIterator;
+import org.junit.jupiter.api.Test;
+
+import datawave.query.attributes.Document;
+import datawave.query.iterator.aggregation.DocumentData;
+
+/**
+ * Test for the {@link IndexOnlyKeyToDocumentData} which operates on TF columns
+ */
+public class IndexOnlyKeyToDocumentDataTest {
+
+    // the document key for the IndexOnlyKeyToDocumentData is different
+    private final Key documentKey = new Key("20230114_17", "tf", "datatype\0uid");
+    private final Key childKey = new Key("20230114_17", "tf", "datatype\0uid.1");
+    private final Key grandChildKey = new Key("20230114_17", "tf", "datatype\0uid.2.7");
+
+    private final Value value = new Value();
+
+    @Test
+    public void testEventData_singleValue() {
+        String field = "FIELD_A";
+        drive(field, documentKey, getSource(), 1);
+    }
+
+    @Test
+    public void testEventData_multiValue() {
+        String field = "FIELD_B";
+        drive(field, documentKey, getSource(), 2);
+    }
+
+    @Test
+    public void testEventData_noData() {
+        String field = "FIELD_K";
+        drive(field, documentKey, getSource(), 0);
+    }
+
+    @Test
+    public void testTLDData_parent_singleValue() {
+        String field = "FIELD_A";
+        drive(field, documentKey, getTLDSource(), 1);
+    }
+
+    @Test
+    public void testTLDData_child_singleValue() {
+        String field = "FIELD_A";
+        drive(field, childKey, getTLDSource(), 1);
+    }
+
+    @Test
+    public void testTLDData_grandchild_singleValue() {
+        String field = "FIELD_X";
+        drive(field, grandChildKey, getTLDSource(), 1);
+    }
+
+    @Test
+    public void testTLDData_parent_multiValue() {
+        String field = "FIELD_B";
+        drive(field, documentKey, getTLDSource(), 2);
+    }
+
+    @Test
+    public void testTLDData_child_multiValue() {
+        String field = "FIELD_B";
+        drive(field, childKey, getTLDSource(), 2);
+    }
+
+    /**
+     * Drive the aggregation and assert expected size
+     *
+     * @param field
+     *            the field
+     * @param documentKey
+     *            the document key
+     * @param source
+     *            the source
+     * @param expectedSize
+     *            the expected size post-aggregation
+     */
+    private void drive(String field, Key documentKey, SortedKeyValueIterator<Key,Value> source, int expectedSize) {
+        IndexOnlyKeyToDocumentData data = new IndexOnlyKeyToDocumentData(getRange(), field, source);
+        Map.Entry<DocumentData,Document> result = data.apply(getEntry(documentKey));
+
+        int dataSize = result.getKey().getData().size();
+        assertEquals(expectedSize, dataSize);
```

Comment:
Should we be checking the contents of the map?

Reply:
done